### PR TITLE
Add Three-Way-Quicksort and Dutch National Flag Sort in C++

### PR DESCRIPTION
**Pull request by Pl4gue**

### CODE REVIEW

File: code/sorting/quick_sort/dutch_national_flag.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            break;
+        case 2:
+            swap(&a[mid], &a[hi--]);
+            break;
+        }
+    }
+}
+ 
+/* UTILITY FUNCTIONS */
+void swap(int *a, int *b)
```
Comment (arnavb) [inline code/sorting/quick_sort/dutch_national_flag.cpp]: Use Pass by reference; This is C++

File: code/sorting/quick_sort/dutch_national_flag.cpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+{
+    int temp = *a;
+    *a = *b;
+    *b = temp;
+}
+ 
+void printarr(int a[], int n)
```
Comment (arnavb) [inline code/sorting/quick_sort/dutch_national_flag.cpp]: Consider using vectors, not arrays.

File: code/sorting/quick_sort/threeWay_quick_sort.cpp [hunk 3]
```diff
@@ -0,0 +1,91 @@
+/* Part of Cosmos by OpenGenus Foundation */
```
Comment (arnavb) [inline code/sorting/quick_sort/threeWay_quick_sort.cpp]: Do the same changes as previous file

File: code/sorting/quick_sort/dutch_national_flag.cpp (hunk 1 above)
Comment (Pl4gue) [reply code/sorting/quick_sort/dutch_national_flag.cpp]: Done

File: code/sorting/quick_sort/dutch_national_flag.cpp (hunk 2 above)
Comment (Pl4gue) [reply code/sorting/quick_sort/dutch_national_flag.cpp]: Done

File: code/sorting/quick_sort/threeWay_quick_sort.cpp (hunk 3 above)
Comment (Pl4gue) [reply code/sorting/quick_sort/threeWay_quick_sort.cpp]: Done

**Comment by Pl4gue** [comment]
Fixed all mentioned issues. Could you please merge now? :)

**Comment by Pl4gue** [comment]
I also cleaned up the code a little bit.

**Comment by arnavb** [comment]
Fix the merge conflict, and then @AdiChat will merge this.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**